File: Number_of_Good_Pairs.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var numIdenticalPairs = function(nums) {
    const pairs = {};
    let count = 0;

    for (let i = 0; i < nums.length; i++) {
        if (nums[i] in pairs) {
            count += pairs[nums[i]];
        }
        
        pairs[nums[i]] = (pairs[nums[i]] || 0) + 1;
    }

    return count;    
};

File: Number_of_Good_Pairs.cpp
class Solution {
public:
    int numIdenticalPairs(vector<int>& nums) {
        unordered_map<int, int> pairs;
        int count = 0;

        for (int i = 0; i < nums.size(); i++) {
            if (pairs.find(nums[i]) != pairs.end()) {
                count += pairs[nums[i]];
            }
            pairs[nums[i]] = pairs[nums[i]] + 1;
        }

        return count;        
    }
};
